### ISSUE TITLE
Unstable test ddl `mock update version error`

### ISSUE THREAD

**Issue by aytrack** [pingcap/tidb#29756]
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
https://ci.pingcap.net/blue/organizations/jenkins/tidb_ghpr_coverage/detail/tidb_ghpr_coverage/1180/pipeline

```bash
2021-11-13T19:44:01.221Z] [2021/11/14 03:39:18.090 +08:00] [ERROR] [ddl_worker.go:691] ["[ddl] run DDL job error"] [worker="worker 223, tp general"] [error="mock update version error"] [errorVerbose="mock update version error\ngithub.com/pingcap/tidb/ddl.onTruncateTable\n\t/home/jenkins/agent/workspace/tidb_ghpr_coverage/go/src/github.com/pingcap/tidb/ddl/table.go:691\ngithub.com/pingcap/tidb/ddl.(*worker).runDDLJob\n\t/home/jenkins/agent/workspace/tidb_ghpr_coverage/go/src/github.com/pingcap/tidb/ddl/ddl_worker.go:787\ngithub.com/pingcap/tidb/ddl.(*worker).handleDDLJobQueue.func1\n\t/home/jenkins/agent/workspace/tidb_ghpr_coverage/go/src/github.com/pingcap/tidb/ddl/ddl_worker.go:542\ngithub.com/pingcap/tidb/kv.RunInNewTxn\n\t/home/jenkins/agent/workspace/tidb_ghpr_coverage/go/src/github.com/pingcap/tidb/kv/txn.go:47\ngithub.com/pingcap/tidb/ddl.(*worker).handleDDLJobQueue\n\t/home/jenkins/agent/workspace/tidb_ghpr_coverage/go/src/github.com/pingcap/tidb/ddl/ddl_worker.go:498\ngithub.com/pingcap/tidb/ddl.(*worker).start\n\t/home/jenkins/agent/workspace/tidb_ghpr_coverage/go/src/github.com/pingcap/tidb/ddl/ddl_worker.go:198\nruntime.goexit\n\t/usr/local/go/src/runtime/asm_amd64.s:1371"]

[2021-11-13T19:44:01.222Z] [2021/11/14 03:39:18.092 +08:00] [INFO] [ddl_worker.go:571] ["[ddl] run DDL job failed, sleeps a while then retries it."] [worker="worker 223, tp general"] [waitTime=1µs] [error="mock update version error"] [errorVerbose="mock update version error\ngithub.com/pingcap/tidb/ddl.onTruncateTable\n\t/home/jenkins/agent/workspace/tidb_ghpr_coverage/go/src/github.com/pingcap/tidb/ddl/table.go:691\ngithub.com/pingcap/tidb/ddl.(*worker).runDDLJob\n\t/home/jenkins/agent/workspace/tidb_ghpr_coverage/go/src/github.com/pingcap/tidb/ddl/ddl_worker.go:787\ngithub.com/pingcap/tidb/ddl.(*worker).handleDDLJobQueue.func1\n\t/home/jenkins/agent/workspace/tidb_ghpr_coverage/go/src/github.com/pingcap/tidb/ddl/ddl_worker.go:542\ngithub.com/pingcap/tidb/kv.RunInNewTxn\n\t/home/jenkins/agent/workspace/tidb_ghpr_coverage/go/src/github.com/pingcap/tidb/kv/txn.go:47\ngithub.com/pingcap/tidb/ddl.(*worker).handleDDLJobQueue\n\t/home/jenkins/agent/workspace/tidb_ghpr_coverage/go/src/github.com/pingcap/tidb/ddl/ddl_worker.go:498\ngithub.com/pingcap/tidb/ddl.(*worker).start\n\t/home/jenkins/agent/workspace/tidb_ghpr_coverage/go/src/github.com/pingcap/tidb/ddl/ddl_worker.go:198\nruntime.goexit\n\t/usr/local/go/src/runtime/asm_amd64.s:1371"]
```

**Comment by Defined2014** [comment]
failed msg
```
[2021-11-13T19:43:55.375Z] FAIL: attributes_sql_test.go:391: testDBSuite8.TestCreateWithSameName
[2021-11-13T19:43:55.375Z] 
[2021-11-13T19:43:55.375Z] attributes_sql_test.go:450:
[2021-11-13T19:43:55.375Z]     c.Assert(len(rows), Equals, 3)
[2021-11-13T19:43:55.375Z] ... obtained int = 4
[2021-11-13T19:43:55.375Z] ... expected int = 3
[2021-11-13T19:43:55.375Z] 
```

**Comment by Defined2014** [comment]
This issue may be the same as #29734, failed in same test `testDBSuite8.TestCreateWithSameName`

**Comment by Defined2014** [comment]
Maybe fixed by #28898, close the issue first.
